plugins/org.jkiss.dbeaver.erd.ui/src/org/jkiss/dbeaver/erd/ui/notations/ERDNotationRegistry.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2023 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.erd.ui.notations;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IConfigurationElement;
+import org.eclipse.core.runtime.IExtensionRegistry;
+import org.eclipse.core.runtime.Platform;
+import org.jkiss.code.NotNull;
+import org.jkiss.dbeaver.Log;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class ERDNotationRegistry {
+
+    private Log log = Log.getLog(ERDNotationRegistry.class);
+    private Map<String, ERDNotationDescriptor> notations = new LinkedHashMap<>();
+    private ERDNotationDescriptor defaultNotation;
+    private static ERDNotationRegistry instance;
+    private static final String EXTENSION_ID = "org.jkiss.dbeaver.erd.ui.notation.style";
+
+    private ERDNotationRegistry(IExtensionRegistry registry) {
+        IConfigurationElement[] cfgElements = registry.getConfigurationElementsFor(EXTENSION_ID);
+        Stream.of(cfgElements).forEach(c -> {
```

Comment:
Stop abusing Stream APIs

```suggestion
        for (IConfigurationElement c : cfgElements) {
```

Reply:
Argue your comment pls : `"Stop abusing Stream APIs"  `

Reply:
Streams create unnecessary noise and slow down code. What's the point of creating a new stream and iterating over its elements rather than using a plain for loop?

Reply:
The java Stream does not store elements and has a functional in nature, this is standard for since Java 8 been released.
A lot of greate articles covering Streams vs For, one of them:
https://blog.knoldus.com/java-streams-vs-loop/

Reply:
Streams are good for a processing pipeline. You filter objects, map them, filter again, sort and collect. If you were to do that using for loops, it would require way more code than using streams. But if we are talking about just iterating over elements of an array, streams are just too much. You instantiate a new Stream object, create a bootstrap method with a callback for the `forEach` method, and then, under the hood, it performs a plain `foreach` loop under the hood. What's the point?

Reply:
Vote for Stream

Reply:
> Using Streams is a way to reach more efficient code, that looks greate and simplify product maintance
Not in that case.